src/Data/Nat/Properties.agda

```diff
@@ -2106,31 +2107,48 @@ n≤′m+n (suc m) n = ≤′-step (n≤′m+n m n)
 -- Properties of _≤″_ and _<″_
 ------------------------------------------------------------------------
 
-m<ᵇn⇒1+m+[n-1+m]≡n : ∀ m n → T (m <ᵇ n) → suc m + (n ∸ suc m) ≡ n
-m<ᵇn⇒1+m+[n-1+m]≡n m n lt = m+[n∸m]≡n (<ᵇ⇒< m n lt)
+-- equivalence of  _≤″_ to _≤_
 
-m<ᵇ1+m+n : ∀ m {n} → T (m <ᵇ suc (m + n))
-m<ᵇ1+m+n m = <⇒<ᵇ (m≤m+n (suc m) _)
+≤⇒≤″ : _≤_ ⇒ _≤″_
+≤⇒≤″ = (_ ,_) ∘ m+[n∸m]≡n
 
-<ᵇ⇒<″ : T (m <ᵇ n) → m <″ n
-<ᵇ⇒<″ {m} {n} leq = less-than-or-equal (m+[n∸m]≡n (<ᵇ⇒< m n leq))
+<⇒<″ : _<_ ⇒ _<″_
+<⇒<″ = ≤⇒≤″
 
-<″⇒<ᵇ : ∀ {m n} → m <″ n → T (m <ᵇ n)
-<″⇒<ᵇ {m} (<″-offset k) = <⇒<ᵇ (m≤m+n (suc m) k)
+≤″⇒≤ : _≤″_ ⇒ _≤_
+≤″⇒≤ (k , refl) = m≤m+n _ k
 
 -- equivalence to the old definition of _≤″_
 
-≤″-proof : ∀ {m n} (le : m ≤″ n) → let less-than-or-equal {k} _ = le in m + k ≡ n
-≤″-proof (less-than-or-equal prf) = prf
+≤″-proof : (le : m ≤″ n) → let k , _ = le in m + k ≡ n
+≤″-proof (_ , prf) = prf
 
--- equivalence to _≤_
+-- yielding analogous proof for _≤_
 
-≤″⇒≤ : _≤″_ ⇒ _≤_
-≤″⇒≤ {zero}  (≤″-offset k) = z≤n {k}
-≤″⇒≤ {suc m} (≤″-offset k) = s≤s (≤″⇒≤ (≤″-offset k))
+m≤n⇒∃[o]m+o≡n : .(m ≤ n) → ∃ λ k → m + k ≡ n
+m≤n⇒∃[o]m+o≡n m≤n = _ , m+[n∸m]≡n (recompute (_ ≤? _) m≤n)
 
-≤⇒≤″ : _≤_ ⇒ _≤″_
-≤⇒≤″ = less-than-or-equal ∘ m+[n∸m]≡n
+-- and a 'guarded' version of monus
+
+guarded-∸ : ∀ n m → .(m ≤ n) → ℕ
+guarded-∸ n m m≤n = let k , _ = m≤n⇒∃[o]m+o≡n m≤n in k
```

Comment:
I dislike having this in `Properties` as an exposed definition, as it's not a property. Can we inline it in the proof below as that's the only place where it is used?

Reply:
OK, fine!